configuration_go/abstr/kubernetes/cmdoption/cmdopt_test.go

```diff
@@ -0,0 +1,144 @@
+package cmdopt_test
+
+import (
+	"testing"
+	"time"
+
+	cmdopt "github.com/observatorium/observatorium/configuration_go/abstr/kubernetes/cmdoption"
+)
+
+type SubStruct struct {
+	SubString string
+}
+
+func (s SubStruct) String() string {
+	return s.SubString
+}
+
+type TestOptions struct {
+	String       string        `opt:"string"`
+	Int          int           `opt:"int"`
+	Float        float64       `opt:"float"`
+	Bool         bool          `opt:"bool"`
+	Duration     time.Duration `opt:"duration"`
+	Sub          SubStruct     `opt:"sub"`
+	NoValue      bool          `opt:"no-value,noval"`
+	Repeat       []string      `opt:"repeat"`
+	SingleHyphen int           `opt:"single,single-hyphen"`
+}
+
+func TestCmdOptions(t *testing.T) {
```

Comment:
Sure, I'm always up for good coverage. Regarding the tests you propose, I am excluding some because there seem to be some confusion about the tags purpose because of how I named them in the tests. 
The first tag value is just the name that is used to generate the arg, it is not interpreted.